src/module.c

```diff
@@ -3022,14 +3045,15 @@ int RM_HashSet(RedisModuleKey *key, int flags, ...) {
             if (((flags & REDISMODULE_HASH_XX) && !exists) ||
                 ((flags & REDISMODULE_HASH_NX) && exists))
             {
+                errno = exists ? EEXIST : ENOENT;
                 if (flags & REDISMODULE_HASH_CFIELDS) decrRefCount(field);
                 continue;
             }
         }
 
         /* Handle deletion if value is REDISMODULE_HASH_DELETE. */
         if (value == REDISMODULE_HASH_DELETE) {
-            updated += hashTypeDelete(key->value, field->ptr);
+            count += hashTypeDelete(key->value, field->ptr);
```

Comment:
this one can return 0, in which case the return value can be less than requested and we didn't set errno